plugins/documents/app/controllers/documents_controller.rb

```diff
@@ -0,0 +1,63 @@
+require 'filemagic'
+
+class DocumentsController < ApplicationController
+  before_filter -> { require_plugin_enabled FoodsoftDocuments }
+
+  def index
+    if params["sort"]
+      sort = case params["sort"]
+               when "name" then "name"
+               when "created_at" then "created_at"
+               when "name_reverse" then "name DESC"
+               when "created_at_reverse" then "created_at DESC"
+             end
+    else
+      sort = "name"
+    end
+
+    @documents = Document.page(params[:page]).per(@per_page).order(sort)
+  end
+
+  def new
+    @document = Document.new
+  end
+
+  def create
+    @document = Document.new
+    @document.data = params[:document][:data].read
+    @document.mime = FileMagic.new(FileMagic::MAGIC_MIME).buffer(@document.data)
```

Comment:
i do not think that whitelisting will have any value. As long as we do not include a "malicious code checker" (which does not exist), more or less every file (HTML, PDF, DOC) can contain bad code. Then the whitelist contains only TXT, which makes the whole documents useless again. IMHO since documents (a) are an optional feature and (b) can only be accessed by a authenticated user, this is not such a big problem. IMHO the important part is, that an attacker can not get any information from the server. When/if an other user will download a mad file can not be avoided to 100%.